### PR TITLE
Internal metrics, only if metrics are enabled

### PR DESCRIPTION
**Pull request by atoulme**
Do not create a pipeline to send internal metrics if no metrics are enabled.

### CODE REVIEW

**Review by dloucasfx — COMMENTED** [review]
So the idea was to keep internal metrics intact, even if `o11yMetricsEnabled` is true
This was discussed in https://github.com/signalfx/splunk-otel-collector-chart/issues/520 and internally.
The fix was to treat internal metrics in the backend as bundled/free.
I don't think we want to remove the whole pipeline

**Review by dloucasfx — APPROVED** [review]

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl [hunk 1]
```diff
@@ -239,7 +239,7 @@ service:
         {{- end }}
     {{- end }}
 
-    {{- if or (eq (include "splunk-otel-collector.splunkO11yEnabled" $) "true") (eq (include "splunk-otel-collector.platformMetricsEnabled" $) "true") }}
+    {{- if and (or (eq (include "splunk-otel-collector.splunkO11yEnabled" $) "true") (eq (include "splunk-otel-collector.platformMetricsEnabled" $) "true")) (or (eq (include "splunk-otel-collector.splunkO11yEnabled" .) "true") (eq (include "splunk-otel-collector.platformMetricsEnabled" $) "true")) }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:242]: Why is this change needed? looks like both and argument are the same, no?

Comment (atoulme) [reply helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:242]: Oh they are the same now. I missed that. I will fix.

Comment (atoulme) [reply helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:242]: see https://github.com/signalfx/splunk-otel-collector-chart/pull/544
